src/pybind/mgr/cephadm/module.py

```diff
@@ -1866,6 +1895,16 @@ def _create_daemon(self,
                         self.log.warning(msg)
                         return msg
 
+            if daemon_spec.daemon_type == 'haproxy':
+                haspec = cast(HA_RGWSpec, daemon_spec.spec)
+                if haspec.haproxy_container_image:
+                    image = haspec.haproxy_container_image
+
+            if daemon_spec.daemon_type == 'keepalived':
+                haspec = cast(HA_RGWSpec, daemon_spec.spec)
+                if haspec.keepalived_container_image:
+                    image = haspec.keepalived_container_image
```

Comment:
can we please avoid adding those `if daemon_type == foobar? hacks here please? I really invested a lot of time in cleaning up this here. having those hacks creeping is super painful.

Reply:
I agree this is a problem but with haproxy and keepalived being the 4th and 5th daemon types (container, osd and cephadm-exporter as well) to now need something like this (special daemon-specific operations during _create_daemon) it feels like there needs to be some generic way to allow something like this that doesn't clutter up the function. I personally would rather that refactor be its own PR rather than included here with this already being so large.